### PR TITLE
Add Sami's LTO/CFI tree

### PR DESCRIPTION
**Pull request by nathanchance**
Closes: https://github.com/ClangBuiltLinux/continuous-integration2/issues/69

### CODE REVIEW

**Review by nickdesaulniers — APPROVED** [review]
Can you add new badges to the readme for these trees?  Also, it seems like this is strictly only ThinLTO? That's fine, but it might be nice to make that more obvious than just the name of the config file?

**Review by samitolvanen — APPROVED** [review]

**Comment by nathanchance** [comment]
> Can you add new badges to the readme for these trees?

Sure.

> Also, it seems like this is strictly only ThinLTO? That's fine, but it might be nice to make that more obvious than just the name of the config file?

Did you have any particular idea in mind? The logic for naming the workflows is to just pull from the options we fed to TuxSuite. I could move the configuration options into `generator.yml` rather than having the `kernel/configs/` files in Sami's tree but I figured that was a bit of a cleaner solution because then if the configuration options get renamed or there are other dependency configuration options that get pulled in, we can just update those config files rather than `generator.yml`. Crazy solution I suppose would be pulling down the `kernel/configs/` files and dumping them into the workflow name but that seems like overkill :)

File: generator.yml [hunk 1]
```diff
@@ -245,6 +245,21 @@ builds:
   - {<< : *x86_64_cut,    << : *android-4_14,     << : *lld,             boot: true,  llvm_version: *llvm_tot}
   - {<< : *arm64_cut,     << : *android-4_9,      << : *clang,           boot: true,  llvm_version: *llvm_tot}
   - {<< : *x86_64_cut,    << : *android-4_9,      << : *clang,           boot: true,  llvm_version: *llvm_tot}
+  #############
+  #  LTO/CFI  #
+  #############
+  - {<< : *arm64,        << : *lto-cfi,          << : *llvm_full,        boot: true,  llvm_version: *llvm_tot}
+  - {<< : *arm64_lto,    << : *lto-cfi,          << : *llvm_full,        boot: true,  llvm_version: *llvm_tot}
```
Comment (nickdesaulniers) [inline generator.yml:252]: for example, when I see this anchor, I think "full LTO or ThinLTO" then have to go look at the anchor. not a big deal though.

Comment (nathanchance) [reply generator.yml:252]: Ah, I see. Fair point although I am not sure full LTO is worth testing through this matrix. Something to double back on if it is relevant in the future.

**Pull request merged**